standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreUtils.java

```diff
@@ -1143,15 +1143,26 @@ public static TableName getTableNameFor(Table table) {
   /**
    * Because TABLE_NO_AUTO_COMPACT was originally assumed to be NO_AUTO_COMPACT and then was moved
    * to no_auto_compact, we need to check it in both cases.
+   * Check the database level no_auto_compact , if present it is given priority else table level no_auto_compact is considered.
    */
-  public static boolean isNoAutoCompactSet(Map<String, String> parameters) {
-    String noAutoCompact =
-            parameters.get(hive_metastoreConstants.TABLE_NO_AUTO_COMPACT);
+  public static boolean isNoAutoCompactSet(Map<String, String> dbParameters, Map<String, String> tblParameters) {
+    String dbNoAutoCompact = getNoAutoCompact(dbParameters);
+    if (dbNoAutoCompact == null) {
+      LOG.debug("Using table configuration '" + hive_metastoreConstants.TABLE_NO_AUTO_COMPACT + "' for compaction");
+      String noAutoCompact = getNoAutoCompact(tblParameters);
+      return noAutoCompact != null && noAutoCompact.equalsIgnoreCase("true");
+    }
+    LOG.debug("Using database configuration '" + hive_metastoreConstants.TABLE_NO_AUTO_COMPACT + "' for compaction");
+    return dbNoAutoCompact.equalsIgnoreCase("true");
+  }
+
+  // Get no_auto_compact property by checking in both lower and upper cases
+  private static String getNoAutoCompact(Map<String, String> parameters) {
```

Comment:
can we return boolean right away?

Reply:
Actually no ., when db level NO_AUTO_COMPACT is not set which is null(so i need null to check), we consider the table level ., this method is common to get config for both db and table level .,